src/validators/pattern.ts

```diff
@@ -0,0 +1,51 @@
+import { FieldValidationFunctionSync } from '../model';
+
+export const VALIDATOR_TYPE = 'PATTERN';
+const DEFAULT_MESSAGE = 'Please provide a valid format.';
```

Comment:
Missing 

```
export setErrorMessage
let DEFAULT_Message
```

Reply:
I think same issue on required

Reply:
I think setErrorMessage method is not necessary. We can provide message from validation schema

Reply:
The idea with this global setErrorMessage is the ability to change the text once in the application, it can be of great help for internazionalization.

We have two use cases:
  - As a developer I want to replace a default text with my text for any validator of type X.
  - As a developer I want to replace a default text for just one instance of a validator defined in the validation schema.

Examples:

- I want to port to spanish a nif validation, I would make something like

app level:
```javascript
import {setErrorMessage} from './nif-validator';

setErrorMessage('El nif no es valido');
```

- I want to indicate for a generic pattern validator that for the dni field case it should say Invalid DNI (in validation schema)

```
{
  validation: RegEx
  message: 'Invalid DNI'
}
```